lib/pattern.ml

```diff
@@ -16,6 +16,12 @@ let expand_float ~loc s = [%pat? `Float [%p Ast_builder.Default.pfloat ~loc s]]
 
 let expand_var ~loc var = Ast_builder.Default.ppat_var ~loc var
 
+let expand_anti_quotation ~ppat_loc = function
+  | PPat (ppat, _) -> ppat
+  | PStr _
+  | PSig _
+  | PTyp _ -> Raise.bad_expr_antiquotation_payload ~loc:ppat_loc
```

Comment:
Would you mind adding a `bad_pat_antiquotation_payload` function to `Raise` as I believe it requires a slightly different error message. You can probably just copy the message from the expression one and replace `single expression` by `pattern`!

Reply:
If you also feel like adding a test for the error case as well it should be documented in `CONTRIBUTING.md` it would be wonderful but if you don't, don't bother I'll take care of it later!

Reply:
True, completely missed that part. Updated the function and added a test.